File: WikiCodeParser/Tags/WikiCreditTag.cs
using WikiCodeParser.Models;
using WikiCodeParser.Nodes;

namespace WikiCodeParser.Tags
{
    public class WikiCreditTag : Tag
    {
        public WikiCreditTag()
        {
            Token = null;
            Element = "";
        }

        public override bool Matches(State state, string token)
        {
            var peekTag = state.Peek(8);
            var pt = state.PeekTo("]");
            return peekTag == "[credit:" && pt != null && pt.Length > 8 && !pt.Contains("\n");
        }

        public override INode Parse(Parser parser, State state, string scope)
        {
            var index = state.Index;
            if (state.Next() != '[')
            {
                state.Seek(index, true);
                return null;
            }

            var str = state.ScanTo("]");

            if (state.Next() != ']')
            {
                state.Seek(index, true);
                return null;
            }

            var credit = new WikiRevisionCredit
            {
                Type = WikiRevisionCredit.TypeCredit
            };

            var sections = str.Split('|');
            foreach (var section in sections)
            {
                var spl = section.Split(new[] {':'}, 2);
                var key = spl[0];
                var val = spl.Length > 1 ? spl[1] : "";
                switch (key) {
                    case "credit":
                        credit.Description = val;
                        break;
                    case "user":
                        if (int.TryParse(val, out var uid)) credit.UserID = uid;
                        break;
                    case "name":
                        credit.Name = val;
                        break;
                    case "url":
                        credit.Url = val;
                        break;
                }
            }

            state.SkipWhitespace();
            return new MetadataNode("WikiCredit", credit);
        }
    }
}

File: php/src/Tags/WikiCreditTag.php
<?php

namespace LogicAndTrick\WikiCodeParser\Tags;

use LogicAndTrick\WikiCodeParser\Models\WikiRevisionCredit;
use LogicAndTrick\WikiCodeParser\Nodes\INode;
use LogicAndTrick\WikiCodeParser\Nodes\MetadataNode;
use LogicAndTrick\WikiCodeParser\ParseData;
use LogicAndTrick\WikiCodeParser\Parser;
use LogicAndTrick\WikiCodeParser\State;
use LogicAndTrick\WikiCodeParser\TagParseContext;

class WikiCreditTag extends Tag
{
    public function __construct()
    {
        parent::__construct();
        $this->token = null;
        $this->element = '';
    }

    public function Matches(State $state, ?string $token, TagParseContext $context): bool
    {
        $peekTag = $state->Peek(8);
        $pt = $state->PeekTo(']');
        return $peekTag == '[credit:' && $pt != null && strlen($pt) > 8 && !str_contains($pt, "\n");
    }

    public function Parse(Parser $parser, ParseData $data, State $state, string $scope, TagParseContext $context): INode|null
    {
        $index = $state->index;
        if ($state->Next() != '[') {
            $state->Seek($index, true);
            return null;
        }

        $str = $state->ScanTo(']');

        if ($state->Next() != ']') {
            $state->Seek($index, true);
            return null;
        }

        $credit = new WikiRevisionCredit();
        $credit->Type = WikiRevisionCredit::TypeCredit;

        $sections = explode('|', $str);
        foreach ($sections as $section) {
            $spl = explode(':', $section);
            $key = $spl[0];
            $val = count($spl) > 1 ? implode(':', array_slice($spl, 1)) : '';
            switch ($key) {
                case 'credit':
                    $credit->Description = $val;
                    break;
                case 'user':
                    $credit->UserID = intval($val, 10) ?? null;
                    break;
                case 'name':
                    $credit->Name = $val;
                    break;
                case 'url':
                    $credit->Url = $val;
                    break;
            }
        }

        $state->SkipWhitespace();
        return new MetadataNode('WikiCredit', $credit);
    }
}